document_merge_service/api/unoconv.py

```diff
@@ -106,13 +109,38 @@ def run_fork_safe(
     return CompletedProcess(process.args, retcode, stdout, stderr)
 
 
-def run(cmd):
-    return run_fork_safe(
-        [str(arg) for arg in cmd],
+def run(cmd, unshare=True):
+    # Run libreoffice in isolation. If the main broker of libreoffice locks up,
+    # all following calls to unoconv will hang as well. By unsharing the mount namespace,
+    # we get a copy of all the mounts, but we can change them without affecting the
+    # original. So unoconv will never connect to a main broker, and allways start a
+    # new instance. Then run_fork_safe will terminate the process and all the children
+    # of unoconv.
+    #
+    # I think masking /tmp is enough, but we can use this technique for other paths if
+    # needed.
+    if unshare:
+        shell = [
+            "unshare",
+            "--map-root-user",
+            "--ipc",
+            "--mount",
+            "sh",
+        ]
```

Comment:
I remove the pid namespace. It means that libreoffice process can't see each other, but it also means that we start with pid 1 again. And I am scared of the implications.